src/Microsoft.OData.Core/Metadata/MetadataUtils.cs

```diff
@@ -171,25 +171,17 @@ internal static IList<IEdmOperation> CalculateBindableOperationsForType(IEdmType
                 throw new ODataException(Strings.MetadataUtils_CalculateBindableOperationsForType(bindingType.FullTypeName()), exc);
             }
 
-            List<IEdmOperation> operationsFound = new List<IEdmOperation>();
-            foreach (IEdmOperation operation in operations)
-            {
-                if (!operation.IsBound)
-                {
-                    throw new ODataException(Strings.EdmLibraryExtensions_UnBoundOperationsFoundFromIEdmModelFindMethodIsInvalid(operation.Name));
-                }
+            IList<IEdmOperation> operationsFound = operations as IList<IEdmOperation>;
 
-                if (operation.Parameters.FirstOrDefault() == null)
-                {
-                    throw new ODataException(Strings.EdmLibraryExtensions_NoParameterBoundOperationsFoundFromIEdmModelFindMethodIsInvalid(operation.Name));
-                }
+            if(operationsFound != null)
+            {
+                return operationsFound;
+            }
 
-                IEdmOperationParameter bindingParameter = operation.Parameters.FirstOrDefault();
-                IEdmType resolvedBindingType = edmTypeResolver.GetParameterType(bindingParameter).Definition;
-                if (resolvedBindingType.IsAssignableFrom(bindingType))
-                {
-                    operationsFound.Add(operation);
-                }
+            operationsFound = new List<IEdmOperation>();

```

Comment:
>List<IEdmOperation>() [](start = 34, length = 21)

you can just pass operations into constructor: operationsFound = new List<IEdmOperation>(operations);